### ISSUE TITLE
[BUG]: 

### ISSUE THREAD

**Issue by arleeco** [pycaret/pycaret#3456]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

Running setup function on google colaboratory with parameter preprocess set on False, I obtain a Transformed data shape bigger than original data shape in the number of rows.

I wold like to know why and how can I just use my validation set

### Reproducible Example

```python
!pip install --upgrade pycaret
!pip install shap

import pandas as pd
import plotly.express as px
from pycaret.classification import *
from sklearn.model_selection import train_test_split
import shap 

X_train = pd.read_csv('X_train.csv')
X_val = pd.read_csv('X_val.csv')

y_train = X_train['target']
y_val = X_val['target']

numeric_columns = X_train.select_dtypes(include=['float', 'int']).columns.tolist()
numeric_columns = [column for column in numeric_columns if column != 'target']

clf = setup(data = X_train, preprocess = False,  target='conspiracy', session_id=123, numeric_features = numeric_columns, use_gpu = True, test_data = X_val)
```

### Expected Behavior

| Description | Value
Session id | 123
Target | target
Target typ | Binary
Original data shape | (10823, 506)
Transformed data shape | (10823, 506)
Transformed train set shape | (10281, 506)
Transformed test set shape | (542, 506)
Numeric features | 505

### Actual Results

```python-traceback
| Description | Value
Session id | 123
Target | target
Target typ | Binary
Original data shape | (10823, 506)
Transformed data shape | (11907, 506)
Transformed train set shape | (10823, 506)
Transformed test set shape | (1084, 506)
Numeric features | 505
```

### Installed Versions

<details>
PyCaret required dependencies:
                 pip: 22.0.4
          setuptools: 67.6.1
             pycaret: 3.0.0
             IPython: 7.34.0
          ipywidgets: 7.7.1
                tqdm: 4.65.0
               numpy: 1.22.4
              pandas: 1.4.4
              jinja2: 3.1.2
               scipy: 1.10.1
              joblib: 1.2.0
             sklearn: 1.2.2
                pyod: 1.0.9
            imblearn: 0.10.1
   category_encoders: 2.6.0
            lightgbm: 3.3.5
               numba: 0.56.4
            requests: 2.27.1
          matplotlib: 3.7.1
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.13.1
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.16.1
               tbats: 1.1.2
            pmdarima: 2.0.3
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: 0.41.0
           interpret: Not installed
                umap: Not installed
    pandas_profiling: 3.2.0
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
             xgboost: 1.7.4
            catboost: Not installed
              kmodes: Not installed
             mlxtend: 0.14.0
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: 0.2.7
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
               fugue: Not installed
           streamlit: Not installed
             prophet: 1.1.2
</details>

**Comment by ngupta23** [comment]
```
Original data shape | (10823, 506)
Transformed data shape | (10823, 506)
Transformed train set shape | (10281, 506)
Transformed test set shape | (542, 506)
```

Transformed data does not seem to be bigger than the original data.

**Comment by ngupta23** [comment]
Closing this issue since it does not seem to be a problem. Feel free to comment and reopen if you have any feedback.
